### PR TITLE
Adding one restore test that checks for expected exceptions

### PR DESCRIPTION
**Pull request by scbedd**
During integration I sometimes see _nothing_ and I'm trying to figure out the scenarios that cause that.

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationRestoreTests.cs [hunk 1]
```diff
@@ -190,5 +192,68 @@ public async Task Scenario3(string inputJson)
...
+
+                if (!string.IsNullOrEmpty(gitProcessException))
+                {
+                    var assertion = await Assert.ThrowsAsync<HttpException>(async () =>
+                    {
+                        await _defaultStore.Restore(jsonFileLocation);
+                    });
+                    Assert.Contains(gitProcessException, assertion.Message);
+                }
+                else
```
Comment (JimSuplizio) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationRestoreTests.cs:241]: If not having a tag is supposed to succeed, then that scenario should be separate from the "invalid tag" failure scenario which is supposed to fail. Please create a separate function for the nonexistent tag and verify not only the number of files but the versions in them like the other tests, that are pulling files.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationRestoreTests.cs [hunk 2]
```diff
@@ -190,5 +192,68 @@ public async Task Scenario3(string inputJson)
...
+
+            try
+            {
+                var jsonFileLocation = Path.Join(testFolder, GitStoretests.AssetsJson);
+
+                var parsedConfiguration = await _defaultStore.ParseConfigurationFile(jsonFileLocation);
+
+                if (!string.IsNullOrEmpty(gitProcessException))
+                {
+                    var assertion = await Assert.ThrowsAsync<HttpException>(async () =>
```
Comment (JimSuplizio) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationRestoreTests.cs:235]: You're calling the expected exception message string gitProcessException and we're getting and HttpException, maybe just call it expectedExceptionMessage?

**Comment by scbedd** [comment]
> Thank you for adding these scenarios, there are a couple of minor requests.

Responded to both! Thanks.

**Pull request merged**